android/java/org/chromium/chrome/browser/settings/MediaPreferences.java

```diff
@@ -48,10 +50,25 @@ public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         mPageTitle.set(getString(R.string.prefs_media));
         SettingsUtils.addPreferencesFromResource(this, R.xml.media_preferences);
-    }
 
-    @Override
-    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {}
+        if (!ChromeFeatureList.isEnabled(BraveFeatureList.BRAVE_YOUTUBE_SCRIPT_INJECTOR)
+                || !ChromeFeatureList.isEnabled(BraveFeatureList.BRAVE_BACKGROUND_VIDEO_PLAYBACK)) {
+            ChromeSwitchPreference backgroundVideoPlaybackPref =
+                    findPreference(PREF_BACKGROUND_VIDEO_PLAYBACK);
+            if (backgroundVideoPlaybackPref != null) {
+                backgroundVideoPlaybackPref.setVisible(false);
+            }
+        }
+
+        if (!ChromeFeatureList.isEnabled(BraveFeatureList.BRAVE_YOUTUBE_SCRIPT_INJECTOR)
```

Comment:
this is supposed to be `||`? I'm really confused by these checks and guarding things by two features is not ideal

Reply:
`(!A || !B)` is logically equivalent to `!(A && B)`. 

If one of the two is disabled we need to hide `extraYoutubeControlsPref` preference.
I confirm that, logically, works correctly. 
But if having two features is not ideal I'm happy to rework this part.

Reply:
I understand the logic, but I find the check itself to be confusing so I think it should be encapsulated in a method